pkg/kubelet/cm/dra/types.go

```diff
@@ -38,4 +38,18 @@ type Manager interface {
 	// PodMightNeedToUnprepareResources returns true if the pod with the given UID
 	// might need to unprepare resources.
 	PodMightNeedToUnprepareResources(UID types.UID) bool
+
+	// GetPodDynamicResources gets DynamicResource object from the claimInfo cache.
+	GetPodDynamicResources(pod *v1.Pod, container *v1.Container) ([]*DynamicResource, error)
+}
+
+// DynamicResource contains the fileds to populate to pod resources api
+type DynamicResource struct {
+	ClassName string
+
+	ClaimName string
+
+	ClaimNamespace string
+
+	CDIDevices []string
 }
```

Comment:
Is this what we've done in other components as well? Or have we just reused the equivalent of `podresources.DynamicResources` to accumulate / return this information.

Reply:
no of the provider import the podresources package. (so I wanted to be consistent) 
for device plugin there is translation method between  resource-device to podresourcesapi.ContainerDevices [1]. 
In DRA we have the claim info which is not good representation of the pod dynamic resources this is why I created DynamicResource struct. 

[1] - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/container_manager.go#L211

Reply:
I'd need to look at how we do it for other components, but if what you say is true, I would think we should export a type we already  know about, and leave it up to code in the podresources API to figure out how to translate it into something they can use.

Reply:
I remove the DynamicResource  struct and return claimInfo list for the Pod.